### PR TITLE
Add service instance

### PR DESCRIPTION
**Pull request by mbfrahry**
This PR adds Database service instances to the Terraform Provider
https://docs.oracle.com/en/cloud/paas/database-dbaas-cloud/csdbr/api-Service%20Instances.html

```
make testacc TEST=./opc TESTARGS="-run=TestAccOPCDatabaseServiceInstance"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./opc -v -run=TestAccOPCDatabaseServiceInstance -timeout 120m
=== RUN   TestAccOPCDatabaseServiceInstance_Basic
--- PASS: TestAccOPCDatabaseServiceInstance_Basic (1666.37s)
PASS
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./opc -v -run=TestAccOPCDatabaseServiceInstance_importBasic -timeout 120m
=== RUN   TestAccOPCDatabaseServiceInstance_importBasic
--- PASS: TestAccOPCDatabaseServiceInstance_importBasic (1914.72s)
PASS
```

### CODE REVIEW

File: opc/resource_database_service_instance.go [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+	}
+
+	// Only the PaaS level can have a parameter.
+	if input.Level == database.ServiceInstanceLevelPAAS {
+		input.Parameters = getParameter(d)
+	}
+
+	info, err := client.CreateServiceInstance(&input)
+	if err != nil {
+		return fmt.Errorf("Error creating DatabaseServiceInstance: %s", err)
```
Comment (tombuildsstuff) [inline opc/resource_database_service_instance.go]: Should this be %+v?

File: opc/resource_database_service_instance.go [hunk 2]
```diff
@@ -0,0 +1,594 @@
...
+	d.Set("region", result.Region)
+	d.Set("uri", result.URI)
+	d.Set("shape", result.Shape)
+	d.Set("sid", result.SID)
+	d.Set("sm_plugin_version", result.SMPluginVersion)
+	d.Set("subscription_type", result.SubscriptionType)
+	d.Set("timezone", result.Timezone)
+	d.Set("total_shared_storage", result.TotalSharedStorage)
+	d.Set("version", result.Version)
+	return nil
```
Comment (tombuildsstuff) [inline opc/resource_database_service_instance.go]: We should add a TODO here to set the Parameter and vm_ssh_key fields when available

File: opc/resource_database_service_instance.go [hunk 3]
```diff
@@ -0,0 +1,594 @@
...
+	input := database.DeleteServiceInstanceInput{
+		Name: name,
+	}
+	if err := client.DeleteServiceInstance(&input); err != nil {
+		return fmt.Errorf("Error deleting DatabaseServiceInstance")
+	}
+	return nil
+}
+
+func getParameter(d *schema.ResourceData) []database.Parameter {
```
Comment (tombuildsstuff) [inline opc/resource_database_service_instance.go]: Convention would name this “expandParameter”

File: opc/resource_database_service_instance_test.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		if rs.Type != "opc_database_service_instance" {
+			continue
+		}
+
+		input := database.GetServiceInstanceInput{
+			Name: rs.Primary.Attributes["name"],
+		}
+		if info, err := client.GetServiceInstance(&input); err == nil {
+			return fmt.Errorf("DatabaseServiceInstance %s still exists: %#v", input.Name, info)
+		}
```
Comment (tombuildsstuff) [inline opc/resource_database_service_instance_test.go]: Should we be throwing the “err” value here?

File: opc/resource_database_service_instance_test.go [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    shape = "oc1m"
+    subscription_type = "HOURLY"
+    version = "12.2.0.1"
+    vm_public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC3QxPp0BFK+ligB9m1FBcFELyvN5EdNUoSwTCe4Zv2b51OIO6wGM/dvTr/yj2ltNA/Vzl9tqf9AUBL8tKjAOk8uukip6G7rfigby+MvoJ9A8N0AC2te3TI+XCfB5Ty2M2OmKJjPOPCd6+OdzhT4cWnPOM+OAiX0DP7WCkO4Kx2kntf8YeTEurTCspOrRjGdo+zZkJxEydMt31asu9zYOTLmZPwLCkhel8vY6SnZhDTNSNkRzxZFv+Mh2VGmqu4SSxfVXr4tcFM6/MbAXlkA8jo+vHpy5sC79T4uNaPu2D8Ed7uC3yDdO3KRVdzZCfWHj4NjixdMs2CtK6EmyeVOPuiYb8/mcTybrb4F/CqA4jydAU6Ok0j0bIqftLyxNgfS31hR1Y3/GNPzly4+uUIgZqmsuVFh5h0L7qc1jMv7wRHphogo5snIp45t9jWNj8uDGzQgWvgbFP5wR7Nt6eS0kaCeGQbxWBDYfjQE801IrwhgMfmdmGw7FFveCH0tFcPm6td/8kMSyg/OewczZN3T62ETQYVsExOxEQl2t4SZ/yqklg+D9oGM+ILTmBRzIQ2m/xMmsbowiTXymjgVmvrWuc638X6dU2fKJ7As4hxs3rA1BA5sOt0XyqfHQhtYrL/Ovb1iV+C7MRhKicTyoNTc7oVcDDG0VW785d8CPqttDi50w=="
+    parameter {
+      admin_password = "Test_String7"
+      backup_destination = "NONE"
+      sid = "ORCL"
+      usable_storage = 15
+    }
```
Comment (tombuildsstuff) [inline opc/resource_database_service_instance_test.go]: Can we add a Complete test checking more values?

File: opc/resource_database_service_instance.go [hunk 6]
```diff
@@ -0,0 +1,594 @@
...
+			parameter.SnapshotName = val
+		}
+		if val, ok := attrs["source_service_name"].(string); ok && val != "" {
+			parameter.SourceServiceName = val
+		}
+		if val, ok := attrs["db_demo"].(string); ok && val != "" {
+			addParam := database.AdditionalParameters{
+				DBDemo: database.ServiceInstanceBool(val),
+			}
+			parameter.AdditonalParameters = addParam
```
Comment (tombuildsstuff) [inline opc/resource_database_service_instance.go]: Additional*

File: opc/resource_database_service_instance_test.go [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+		if rs.Type != "opc_database_service_instance" {
+			continue
+		}
+
+		input := database.GetServiceInstanceInput{
+			Name: rs.Primary.Attributes["name"],
+		}
+		if _, err := client.GetServiceInstance(&input); err != nil {
+			return fmt.Errorf("Error retrieving state of DatabaseServiceInstance %s: %s", input.Name, err)
+		}
```
Comment (tombuildsstuff) [inline opc/resource_database_service_instance_test.go]: %+v

File: website/docs/index.html.markdown [hunk 8]
```diff
@@ -45,7 +45,9 @@ The following arguments are supported:
 
 * `endpoint` - (Optional) The API endpoint to use, associated with your Oracle Public Cloud account. This is known as the `REST Endpoint` within the Oracle portal. It can also be sourced from the `OPC_ENDPOINT` environment variable.
 
-* `storage_endpoint` - (Optional) The API endpoint to use, associated with your Oracle Public Cloud account. This is known as the `REST Endpoint` within the Oracle portal. Can also be set via the `OPC_STORAGE_ENDPOINT` environment variable.
+* `storage_endpoint` - (Optional) The API endpoint to use, associated with your Oracle Storage Cloud account. This is known as the `REST Endpoint` within the Oracle portal. Can also be set via the `OPC_STORAGE_ENDPOINT` environment variable.
+
+* `database_endpont` - (Optional) The API endpoint to use, associated with Oracle Database Cloud account. This is known as the `REST Endpoint` within the Oracle portal. Can also be set via the `OPC_DATABASE_ENDPOINT` environment variable.
```
Comment (tombuildsstuff) [inline website/docs/index.html.markdown]: Endpoint*

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 9]
```diff
@@ -0,0 +1,168 @@
+---
+layout: "opc"
+page_title: "Oracle: opc_database_service_instance"
+sidebar_current: "docs-opc-resource-service-instance"
+description: |-
+  Creates and manages a service instance in an OPC identity domain.
+---
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: Does this need a blank line for the dashes to be supported properly?

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+
+# opc\_database\_service\_instance
+
+The ``opc_database_service_instance`` resource creates and manages a service instance in an OPC identity domain.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: Single quotes here

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+```hcl
+resource "opc_database_service_instance" "default" {
+  name        = "service-instance-1"
+  description = "This is a description for an service instance"
+  edition = "EE_EP"
+  level = "PAAS"
+  shape = "oc1m"
+  subscription_type = "HOURLY"
+  version = "12.2.0.1"
+  vm_public_key = "an ssh public key"
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: “A* ssh public key”

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the Service Instance.
+
+* `edition` - (Required) Database edition for the service instance. Possible values are SE, EE, EE_HP, or EE_EP.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: We’d normally quote the values in backticks

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+
+* `level` - (Required) Service level for the service instance. Possible values are BASIC or PAAS.
+
+* `shape` - (Required) Desired compute shape. Possible values are oc3, oc4, oc5, oc6, oc1m, oc2m, oc3m, or oc4m.
+
+* `subscription_type` - (Required) Billing unit. Possible values are HOURLY or MONTHLY.
+
+* `version` - (Required) Oracle Database software version; one of: 12.2.0.1, 12.1.0.2, or 11.2.0.4.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: These should be quoted

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+
+* `name` - (Required) The name of the Service Instance.
+
+* `edition` - (Required) Database edition for the service instance. Possible values are SE, EE, EE_HP, or EE_EP.
+
+* `level` - (Required) Service level for the service instance. Possible values are BASIC or PAAS.
+
+* `shape` - (Required) Desired compute shape. Possible values are oc3, oc4, oc5, oc6, oc1m, oc2m, oc3m, or oc4m.
+
+* `subscription_type` - (Required) Billing unit. Possible values are HOURLY or MONTHLY.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: These should be quoted

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the Service Instance.
+
+* `edition` - (Required) Database edition for the service instance. Possible values are SE, EE, EE_HP, or EE_EP.
+
+* `level` - (Required) Service level for the service instance. Possible values are BASIC or PAAS.
+
+* `shape` - (Required) Desired compute shape. Possible values are oc3, oc4, oc5, oc6, oc1m, oc2m, oc3m, or oc4m.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: Quote all the things

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 16]
```diff
@@ -0,0 +1,168 @@
...
+
+* `vm_public_key` - (Required) Public key for the secure shell (SSH). This key will be used for authentication when connecting to the Database Cloud Service instance using an SSH client.
+
+* `parameter` - (Optional) Additional configuration for a service instance. This set is required if level is PAAS. Parameter is documented below.
+
+* `description` - (Optional) A description of the Service Instance.
+
+Parameter supports the following:
+
+* `admin_password` - (Required) Password for Oracle Database administrator users sys and system. The password must meet the following requirements: Starts with a letter. Is between 8 and 30 characters long. Contains letters, at least one number, and optionally, any number of these special characters: dollar sign ($), pound sign (#), and underscore ( _ )
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: Are other characters rejected?

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 17]
```diff
@@ -0,0 +1,168 @@
...
+
+* `backup_destination` - (Required) Backup Destination. Possible values are BOTH, OSS, NONE.
+
+* `char_set` - (Required) Character Set for the Database Cloud Service Instance. Default value is AL32UTF8.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: We should link to the possible values here

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 18]
```diff
@@ -0,0 +1,168 @@
...
+
+`usable_storage` - (Required) Storage size for data (in GB). Minimum value is 15. Maximum value depends on the backup destination: if BOTH is specified, the maximum value is 1200; if OSS or NONE is specified, the maximum value is 2048.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: Quote these values

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 19]
```diff
@@ -0,0 +1,168 @@
...
+
+* `cloud_storage_container` - (Optional) Name of the Oracle Storage Cloud Service container used to provide storage for your service instance backups. Use the following format to specify the container name: <storageservicename>-<storageidentitydomain>/<containername>
+
+* `cloud_storage_username` - (Optional) Username for the Oracle Storage Cloud Service administrator.
+
+* `cloud_storage_password` - (Optional) Password for the Oracle Storage Cloud Service administrator.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: We should add a comment this is stored in the text in plain text - see the azure SQL resources for an example

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 20]
```diff
@@ -0,0 +1,168 @@
...
+
+* `create_storage_container_if_missing` - (Optional) Specify if the given cloud_storage_container is to be created if it does not already exist. Valid values are true and false. Default value is false.
+
+* `disaster_recovery` - (Optional) Specify if an Oracle Data Guard configuration is created using the Disaster Recovery option or the High Availability option. Valid values are yes and no. Default value is no.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: Should we make this consistent as true/false?

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 21]
```diff
@@ -0,0 +1,168 @@
...
+
+* `failover_database` - Specify if an Oracle Data Guard configuration comprising a primary database and a standby database is created. Valid values are yes and no. Default value is no.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: Same as above

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 22]
```diff
@@ -0,0 +1,168 @@
...
+
+* `golden_gate` - (Optional) Specify if the database should be configured for use as the replication database of an Oracle GoldenGate Cloud Service instance. Valid values are yes and no. Default value is no. You cannot set goldenGate to yes if either isRac or failoverDatabase is set to yes.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: Quotes + the point about true/false

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 23]
```diff
@@ -0,0 +1,168 @@
...
+
+* `ibkup` - (Optional) Specify if the service instance's database should, after the instance is created, be replaced by a database stored in an existing cloud backup that was created using Oracle Database Backup Cloud Service. Valid values are yes and no. Default value is no.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: This feels like it’d be better as a block imo

File: website/docs/r/opc_database_service_instance.html.markdown [hunk 24]
```diff
@@ -0,0 +1,168 @@
...
+Username of the Oracle Cloud user. This parameter is required if ibkup is set to yes.
+
+* `ibkup_cloud_storage_password` - (Optional)
+Password of the Oracle Cloud user specified in ibkup_cloud_storage_user. This parameter is required if ibkup is set to yes.
+
+* `ibkup_database_id` - (Optional) Database id of the database from which the existing cloud backup was created. This parameter is required if ibkup is set to yes.
+
+* `ibkup_decryption_key` - (Optional) Password used to create the existing, password-encrypted cloud backup. This password is used to decrypt the backup. Specify either ibkup_decryption_key or ibkup_wallet_file_content for decrypting the backup. This parameter is required if ibkup is set to yes.
+
+* `ibkup_wallet_file_content` - (Optional) String containing the xsd:base64Binary representation of the cloud backup's wallet file. This wallet is used to decrypt the backup. Specify either ibkup_decryption_key or ibkup_wallet_file_content for decrypting the backup. This parameter is required if ibkup is set to yes.
```
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: In other clouds we handle the base64 encoding if it’s not already base64 encoded

File: opc/config.go [hunk 25]
```diff
@@ -82,6 +85,19 @@ func (c *Config) Client() (*OPCClient, error) {
 		opcClient.storageClient = storageClient
 	}
 
+	if c.DatabaseEndpoint != "" {
+		databaseEndpoint, err := url.ParseRequestURI(c.DatabaseEndpoint)
+		if err != nil {
+			return nil, fmt.Errorf("Invalid database endpoint URI: %s", err)
```
Comment (mbfrahry) [inline opc/config.go]: Done

File: opc/import_database_service_instance_test.go [hunk 26]
```diff
@@ -0,0 +1,34 @@
...
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"parameter", "vm_public_key"},
+			},
```
Comment (mbfrahry) [inline opc/import_database_service_instance_test.go]: I don't believe this matters. We can't update anything about a service instance. Correct me if I'm wrong though

File: opc/resource_database_service_instance.go [hunk 27]
```diff
@@ -0,0 +1,594 @@
...
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(database.ServiceInstanceShapeOC3),
+					string(database.ServiceInstanceShapeOC4),
+					string(database.ServiceInstanceShapeOC5),
+					string(database.ServiceInstanceShapeOC6),
+					string(database.ServiceInstanceShapeOC1M),
+					string(database.ServiceInstanceShapeOC2M),
+					string(database.ServiceInstanceShapeOC3M),
+					string(database.ServiceInstanceShapeOC4M),
```
Comment (mbfrahry) [inline opc/resource_database_service_instance.go]: What do you mean?

File: opc/resource_database_service_instance.go [hunk 28]
```diff
@@ -0,0 +1,594 @@
...
+				ValidateFunc: validation.StringInSlice([]string{
+					string(database.ServiceInstanceSubscriptionTypeHourly),
+					string(database.ServiceInstanceSubscriptionTypeMonthly),
+				}, true),
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
```
Comment (mbfrahry) [inline opc/resource_database_service_instance.go]: What're you thinking here?

File: opc/resource_database_service_instance.go [hunk 29]
```diff
@@ -0,0 +1,594 @@
...
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							Default:  database.ServiceInstanceNo,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(database.ServiceInstanceYes),
+								string(database.ServiceInstanceNo),
+							}, true),
+						},
+						"ibkup": {
```
Comment (mbfrahry) [inline opc/resource_database_service_instance.go]: Product name.

File: opc/resource_database_service_instance.go [hunk 30]
```diff
@@ -0,0 +1,594 @@
...
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Computed: true,
+			},
+			"current_version": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Computed: true,
+			},
+			"dbaasmonitor_url": {
```
Comment (mbfrahry) [inline opc/resource_database_service_instance.go]: Can do

File: opc/resource_database_service_instance.go (hunk 3 above)
Comment (mbfrahry) [inline opc/resource_database_service_instance.go]: Makes sense!

File: opc/resource_database_service_instance.go (hunk 6 above)
Comment (mbfrahry) [inline opc/resource_database_service_instance.go]: Ooo. That's a misspelling in the sdk. Will have to fix there first

File: opc/resource_database_service_instance_test.go (hunk 4 above)
Comment (mbfrahry) [inline opc/resource_database_service_instance_test.go]: We are checking to see if the instance is still there after deletion. If there is no error then we found the instance which is a test failure. Should be good as is

File: opc/resource_database_service_instance_test.go (hunk 5 above)
Comment (mbfrahry) [inline opc/resource_database_service_instance_test.go]: That'd be a good idea.

File: website/docs/r/opc_database_service_instance.html.markdown (hunk 9 above)
Comment (mbfrahry) [inline website/docs/r/opc_database_service_instance.html.markdown]: I'm not sure what you mean.

File: opc/import_database_service_instance_test.go (hunk 26 above)
Comment (tombuildsstuff) [inline opc/import_database_service_instance_test.go]: true, but we're still going to need this when importing a resource

File: website/docs/r/opc_database_service_instance.html.markdown (hunk 10 above)
Comment (mbfrahry) [inline website/docs/r/opc_database_service_instance.html.markdown]: This is the case in all of our docs. Should I change them there too?

File: opc/resource_database_service_instance.go (hunk 27 above)
Comment (tombuildsstuff) [inline opc/resource_database_service_instance.go]: it depends how often new Shape's are added - but whitelisting Machine SKU's can be painful if they're added frequently. In some Providers this means we don't validate the Machine SKU and let the remote cloud return an error if it's invalid (or check a regex instead)

File: website/docs/r/opc_database_service_instance.html.markdown (hunk 16 above)
Comment (mbfrahry) [inline website/docs/r/opc_database_service_instance.html.markdown]: This is it

File: website/docs/r/opc_database_service_instance.html.markdown (hunk 10 above)
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: TIL, seems odd for them to be 2 and not 1 but ¯\_(ツ)_/¯

File: website/docs/r/opc_database_service_instance.html.markdown (hunk 16 above)
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: 👍 as such could we add "any other characters are rejected" to the end of this

File: website/docs/r/opc_database_service_instance.html.markdown (hunk 17 above)
Comment (mbfrahry) [inline website/docs/r/opc_database_service_instance.html.markdown]: Done

File: website/docs/r/opc_database_service_instance.html.markdown (hunk 9 above)
Comment (tombuildsstuff) [inline website/docs/r/opc_database_service_instance.html.markdown]: Sorry, I meant does this render ok when you run the website locally? I've had issues in the past with this not rendering correctly without a blank line between the content and the dashes

File: website/docs/r/opc_database_service_instance.html.markdown (hunk 20 above)
Comment (mbfrahry) [inline website/docs/r/opc_database_service_instance.html.markdown]: Do this as true/false and then convert it to yes/no before we send it to the sdk?

File: website/docs/r/opc_database_service_instance.html.markdown (hunk 24 above)
Comment (mbfrahry) [inline website/docs/r/opc_database_service_instance.html.markdown]: I'll look into adding this

File: opc/resource_database_service_instance.go (hunk 27 above)
Comment (mbfrahry) [inline opc/resource_database_service_instance.go]: I'll take it out then.

File: opc/resource_database_service_instance.go (hunk 28 above)
Comment (mbfrahry) [inline opc/resource_database_service_instance.go]: I'll take it out

File: opc/provider.go [hunk 31]
```diff
@@ -56,6 +56,12 @@ func Provider() terraform.ResourceProvider {
 				DefaultFunc: schema.EnvDefaultFunc("OPC_STORAGE_ENDPOINT", nil),
 				Description: "The HTTP endpoint for Oracle Storage operations.",
 			},
```
Comment (mbfrahry) [inline opc/provider.go]: Is this wrong?

File: website/docs/r/opc_database_service_instance.html.markdown (hunk 16 above)
Comment (mbfrahry) [inline website/docs/r/opc_database_service_instance.html.markdown]: Oh, I misread. Their docs don't say anything about rejecting other characters. I just think this is added security. I'll add it to another test to be sure

**Comment by mbfrahry** [comment]
```
make testacc TEST=./opc TESTARGS="-run=TestAccOPCDatabaseServiceInstance"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./opc -v -run=TestAccOPCDatabaseServiceInstance -timeout 120m
=== RUN   TestAccOPCDatabaseServiceInstance_importBasic
--- PASS: TestAccOPCDatabaseServiceInstance_importBasic (1687.53s)
=== RUN   TestAccOPCDatabaseServiceInstance_Basic
--- PASS: TestAccOPCDatabaseServiceInstance_Basic (1679.59s)
PASS
```

**Comment by mbfrahry** [comment]
Test passing for cloud storage 

```
make testacc TEST=./opc TESTARGS="-run=TestAccOPCDatabaseServiceInstance_CloudStorage"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./opc -v -run=TestAccOPCDatabaseServiceInstance_CloudStorage -timeout 120m
=== RUN   TestAccOPCDatabaseServiceInstance_CloudStorage
--- PASS: TestAccOPCDatabaseServiceInstance_CloudStorage (1694.61s)
PASS
ok  	github.com/terraform-providers/terraform-provider-opc/opc	1694.635s
```

**Comment by mbfrahry** [comment]
After updating to v0.3.1

```
make testacc TEST=./opc TESTARGS="-run=TestAccOPCDatabaseServiceInstance"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./opc -v -run=TestAccOPCDatabaseServiceInstance -timeout 120m
=== RUN   TestAccOPCDatabaseServiceInstance_importBasic
--- PASS: TestAccOPCDatabaseServiceInstance_importBasic (2191.24s)
=== RUN   TestAccOPCDatabaseServiceInstance_Basic
--- PASS: TestAccOPCDatabaseServiceInstance_Basic (2367.87s)
=== RUN   TestAccOPCDatabaseServiceInstance_CloudStorage
--- PASS: TestAccOPCDatabaseServiceInstance_CloudStorage (2309.33s)
PASS
ok  	github.com/terraform-providers/terraform-provider-opc/opc	6868.467s
```

**Pull request merged**